pkgs/dart_tooling_mcp_server/lib/src/mixins/pub_dev_search.dart

```diff
@@ -0,0 +1,265 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:dart_mcp/server.dart';
+import 'package:http/http.dart';
+
+import '../utils/process_manager.dart';
+
+/// Mix this in to any MCPServer to add support for doing searched on pub.dev.
+base mixin PubDevSupport on ToolsSupport, LoggingSupport
+    implements ProcessManagerSupport {
+  @override
+  FutureOr<InitializeResult> initialize(InitializeRequest request) {
+    registerTool(pubDevTool, _runPubDevSearch);
+    return super.initialize(request);
+  }
+
+  /// Implementation of the [pubDevTool].
+  Future<CallToolResult> _runPubDevSearch(CallToolRequest request) async {
+    final query = request.arguments?['search-query'] as String?;
```

Comment:
nit: how about just 'query' instead of 'search-query'

CC @jakemac53 in general, we should probably have some policy for tool names and parameter names to make them consistent. Right now we seem to be using snake case for tool names and camel case for parameter names.